src/module_object.cpp

```diff
@@ -477,45 +477,44 @@ Value ModuleObject::name(Env *env) {
 Value ModuleObject::attr_reader(Env *env, size_t argc, Value *args) {
     for (size_t i = 0; i < argc; i++) {
         Value name_obj = args[i];
-        if (name_obj->type() == Object::Type::String) {
+        if (name_obj->type() == Object::Type::Symbol) {
             // we're good!
-        } else if (name_obj->type() == Object::Type::Symbol) {
-            name_obj = name_obj->as_symbol()->to_s(env);
+        } else if (name_obj->type() == Object::Type::String) {
+            name_obj = SymbolObject::intern(name_obj->as_string()->c_str());
```

Comment:
We should avoid using `c_str()` here and elsewhere because it causes Natalie to not properly handle the null character (`\0`), plus it requires a call to `strlen()`.

You can create a Symbol using the TM::String directly: https://github.com/richardboehme/natalie/blob/d71ab5e75f5b8f7f585bc3a5a2e53282cfdf0b73/include/natalie/symbol_object.hpp#L25

Reply:
I added a method to `StringObject` that returns the `TM::String` reference we have there and updated `attr_reader/writer` to use this method instead of `c_str()`. We save the `strlen` for now and later if `SymbolObject` starts supporting null bytes we do not have to change those lines.